### PR TITLE
Refactor endive and parametric specs projects'

### PR DESCRIPTION
**Pull request by rodrigo7491**
This PR creates two new projects, `endive` and `parametric`, which will contain part of the specs currently present in the `performance` project. This is done to enable parallel runs, as discussed on #52.

### CODE REVIEW

File: parametric/build.sbt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    Bench.Runs(
+      s"run-${encoding}",
+      timelimit = suiteTimeLimit,
+      cmds = lengths.map(checkCmd(encoding, _)),
+      tasks = Seq(Tasks(s"task-${name}-${encoding}", specs))
+    )
+  }
+
+  Bench.Suite(
+    name = s"004parametric-${name}",
```
Comment (shonfeder) [inline parametric/build.sbt]: what does "parametric" refer to here exactly? Given the content and the readme, seems like maybe this project should be named "encoding"?

Comment (rodrigo7491) [reply parametric/build.sbt]: The name came from the [apalache-tests](https://github.com/informalsystems/apalache-tests/tree/master/parametric) repo. We also make encoding comparisons in the "examples" project added in #54, so I don't think "encoding" by itself is very descriptive. I'm open to suggestions.

Comment (shonfeder) [reply parametric/build.sbt]: I guess that if the principle purpose this project's bench marks is to gauge the effectiveness of the encoding, i'd still be inclined to name this "encoding", even if we happen to also vary the encoding in some places in other projects.

re: "parametric", my main question is this: are these specs are testing parametric models in the way that the specs from the apalache-tests repo were? If so, maybe it's a fitting name. If not, maybe we should change it. WDYT? :)

**Review by shonfeder — APPROVED** [review]
LGTM! I'll let you work out the naming question here as you see fit. Let me know if you'd like to talk it over  more.

File: parametric/build.sbt (hunk 1 above)
Comment (rodrigo7491) [reply parametric/build.sbt]: From what I can ascertain, it behaves as the parametric config from apalache-tests, but we added additional runs for the arrays-encoding. In principle, I expect every spec in this repo to (eventually) run with both encodings, that is why I don't want to have a project named "encoding".

Comment (shonfeder) [reply parametric/build.sbt]: Ok. Makes sense to me. Thanks!

**Comment by rodrigo7491** [comment]
@shonfeder There were significant changes to `ProjectUtils`, `parametric/build`, `endive/build`, and `examples/build`, can you have a quick look at those files? The refactoring of `performance/build` will be left for later.

File: examples/build.sbt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  Spec("bakery", "MC3.tla", length = 0, inv = "Inv"),
+  Spec("bakery", "MC3.tla", length = 1, init = "Inv", inv = "Inv"),
+  Spec("changRoberts", "MC4.tla", inv = "Correctness"),
+  Spec("changRoberts", "MC10.tla", inv = "Correctness"),
+  Spec("changRoberts", "MC20.tla", inv = "Correctness"),
+  Spec("paxos", "MC3.tla", length = 13, inv = "V!OneValuePerBallot"),
+  Spec("readersWriters", "MC4.tla", inv = "Safety"),
+  Spec("readersWriters", "MC10.tla", inv = "Safety"),
+  Spec("readersWriters", "MC20.tla", inv = "Safety")
+)
```
Comment (shonfeder) [inline examples/build.sbt:29]: Really nice and concise configuration here!

File: parametric/build.sbt [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+)
+
+lazy val parametricSpecs = Seq(
+  Spec("parametric-specs", "SetAdd.tla", inv = "Inv"),
+  Spec("parametric-specs", "SetAddDel.tla", inv = "Inv"),
+  Spec("parametric-specs", "SetSndRcv.tla", inv = "Inv"),
+  Spec("parametric-specs", "SetSndRcv_NoFullDrop.tla", inv = "Inv"),
+)
+
+def cmdGenGen() = {
```
Comment (shonfeder) [inline parametric/build.sbt]: Is this meant to have `Gen` twice in the name, or should it be

```suggestion
def cmdGen() = {
```

Comment (shonfeder) [inline parametric/build.sbt]:
```suggestion
lazy val cmds =
```

**Review by shonfeder — APPROVED** [review]
LGTM!

File: parametric/build.sbt (hunk 3 above)
Comment (rodrigo7491) [reply parametric/build.sbt]: This generates generators, I will add a comment about it :-)

**Pull request merged**